src/Evolution/Systems/Cce/AnalyticSolutions/GaugeWave.hpp

```diff
@@ -0,0 +1,266 @@
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct PeakTime {
+    using type = double;
+    static constexpr Options::String help{
+        "The time of the peak of the Gaussian envelope."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Duration {
+    using type = double;
+    static constexpr Options::String help{
+        "The characteristic duration of the Gaussian envelope."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  using options = tmpl::list<ExtractionRadius, Mass, Frequency, Amplitude,
+                             PeakTime, Duration>;
+
+  static constexpr Options::String help = {
+      "Analytic solution representing worldtube data for a pure-gauge "
+      "perturbation near a Schwarzschild metric in spherical coordinates"};
+
+  WRAPPED_PUPable_decl_template(GaugeWave);  // NOLINT
+
+  explicit GaugeWave(CkMigrateMessage* /*unused*/) noexcept {}
+
+  // clang doesn't manage to use = default correctly in this case
+  // NOLINTNEXTLINE(modernize-use-equals-default)
+  GaugeWave() noexcept {}
+
+  GaugeWave(double extraction_radius, double mass, double frequency,
+            double amplitude, double peak_time, double duration) noexcept;
+
+  std::unique_ptr<WorldtubeData> get_clone() const noexcept override;
+ private:
+  double coordinate_wave_function(double time) const noexcept;
+
+  double du_coordinate_wave_function(double time) const noexcept;
+
+  double du_du_coordinate_wave_function(double time) const noexcept;
+
+ protected:
+  /// A no-op as the gauge wave solution does not have substantial
+  /// shared computation to prepare before the separate component calculations.
+  void prepare_solution(const size_t /*output_l_max*/,
+                        const double /*time*/) const noexcept override {}
+
+  /*!
+   * \brief Compute the spherical coordinate metric from the closed-form gauge
+   * wave metric.
+   *
+   * \details The transformation of the ingoing Eddington-Finkelstein coordinate
+   * produces metric components in spherical coordinates:
+   *
+   * \f{align*}{
+   * g_{tt} &= \frac{-1}{r^3}\left(r - 2 M\right)
+   * \left(r + \partial_u F(u)\right)^2\\
+   * g_{rt} &= \frac{1}{r^4} \left(r + \partial_u F(u)\right)
+   * \left(2 M r^2 + \left(r - 2 M\right)
+   * \left(r \partial_u F(u) + F(u)\right)\right) \\
+   * g_{rr} &= \frac{1}{r^5} \left(r^2 - r \partial_u F(u) - F(u)\right)
+   * \left(r^3 + 2 M r^2 + \left(r - 2 M\right)
+   * \left(r \partial_u F(u) + F(u)\right)\right) \\
+   * g_{\theta \theta} &= r^2 \\
+   * g_{\phi \phi} &= r^2 \sin^2(\theta),
+   * \f}
+   *
+   * and all other components vanish. Here, \f$F(u)\f$ is defined as
+   *
+   * \f{align*}{
+   * F(u) &= A \sin(\omega u) e^{-(u - u_0)^2 /k^2},\\
+   * \partial_u F(u) &= A \left(-2 \frac{u - u_0}{k^2} \sin(\omega u)
+   * + \omega \cos(\omega u)\right) e^{-(u - u_0)^2 / k^2}.
+   * \f}
+   *
+   * \note The \f$\phi\f$ components are returned in a form for which the
```

Comment:
I don't remember what we did in other solutions. This seems to border up on a `\warning` since using these functions without this knowledge could lead to a... frustrating... experience :P I'll leave it up to you to decide, but something to consider :)

Reply:
I copied this note string from the most recently merged version, but I'm happy to promote them to warnings -- I certainly agree that overlooking this feature can make for a bad time.

Reply:
Okay, totally up to you! Just figured I'd bring it up :) I'm fine either way